assets/dev/scss/frontend/_container.scss

```diff
@@ -58,10 +54,15 @@
 	overflow: var( --overflow );
 	transition: var( --transition );
 
-	// Default top-level containers to be centered.
 	:is( .elementor-section-wrap, [data-elementor-id] ) > & {
+		// Default top-level containers to be centered.
 		--margin-right: auto;
 		--margin-left: auto;
+
+		// Use the `min()` function to fix cases when a top-level Container is 100% width due to `max-width: 100%;`,
+		// or cases when a nested Container overflows the parent Container due to `max-width: 1140px;`.
+		// Capital `M` in order to use the CSS function and not the SCSS one.
+		max-width: Min( 100%, var( --width ) );
 	}
```

Comment:
Limited the max-width to top-level containers.

Reply:
Some clever thinking here, and I've tested this and works.
Nice explanation in the comment.